src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingServiceTest.kt

```diff
@@ -160,82 +163,53 @@ class BookingServiceTest {
-    cancellationReasonRepository = mockCancellationReasonRepository,
-    bedRepository = mockBedRepository,
-    placementRequestRepository = mockPlacementRequestRepository,
-    lostBedsRepository = mockLostBedsRepository,
-    turnaroundRepository = mockTurnaroundRepository,
-    bedMoveRepository = mockBedMoveRepository,
-    premisesRepository = mockPremisesRepository,
-    assessmentRepository = mockAssessmentRepository,
-    notifyConfig = NotifyConfig(),
-    applicationUrlTemplate = "http://frontend/applications/#id",
-    bookingUrlTemplate = "http://frontend/premises/#premisesId/bookings/#bookingId",
-    arrivedAndDepartedDomainEventsDisabled = false,
-  )
-
-  private val bookingServiceWithArrivedAndDepartedDomainEventsDisabled = BookingService(
-    premisesService = mockPremisesService,
-    staffMemberService = mockStaffMemberService,
-    offenderService = mockOffenderService,
-    domainEventService = mockDomainEventService,
-    cas3DomainEventService = mockCas3DomainEventService,
-    cruService = mockCruService,
-    applicationService = mockApplicationService,
-    workingDayCountService = mockWorkingDayCountService,
-    emailNotificationService = mockEmailNotificationService,
-    placementRequestService = mockPlacementRequestService,
-    communityApiClient = mockCommunityApiClient,
-    bookingRepository = mockBookingRepository,
-    arrivalRepository = mockArrivalRepository,
-    cancellationRepository = mockCancellationRepository,
-    confirmationRepository = mockConfirmationRepository,
-    extensionRepository = mockExtensionRepository,
-    dateChangeRepository = mockDateChangeRepository,
-    departureRepository = mockDepartureRepository,
-    nonArrivalRepository = mockNonArrivalRepository,
-    departureReasonRepository = mockDepartureReasonRepository,
-    moveOnCategoryRepository = mockMoveOnCategoryRepository,
-    destinationProviderRepository = mockDestinationProviderRepository,
-    nonArrivalReasonRepository = mockNonArrivalReasonRepository,
-    cancellationReasonRepository = mockCancellationReasonRepository,
-    bedRepository = mockBedRepository,
-    placementRequestRepository = mockPlacementRequestRepository,
-    lostBedsRepository = mockLostBedsRepository,
-    turnaroundRepository = mockTurnaroundRepository,
-    bedMoveRepository = mockBedMoveRepository,
-    premisesRepository = mockPremisesRepository,
-    assessmentRepository = mockAssessmentRepository,
-    notifyConfig = NotifyConfig(),
-    applicationUrlTemplate = "http://frontend/applications/#id",
-    bookingUrlTemplate = "http://frontend/premises/#premisesId/bookings/#bookingId",
-    arrivedAndDepartedDomainEventsDisabled = true,
-  )
+  private val bookingService = createBookingService(arrivedAndDepartedDomainEventsDisabled = false)
+  private val bookingServiceWithArrivedAndDepartedDomainEventsDisabled = createBookingService(arrivedAndDepartedDomainEventsDisabled = true)
 
```

Comment:
would be nice if we can add unit test for the new changes in getBooking() method. Previously it was just a DB call, but now you added couple of service class within the method , so would be helpful if we test the negative flow. I believe you have added integration test to test positive flow.

Reply:
Hi @sudhakar-plr , the logic in the new version of getBooking() is existing code, moved from the PremisesController#premisesPremisesIdBookingsBookingIdGet function (which now calls this new function instead). The only changes I made to the existing code was to make it return AuthroisationActionResult instead of directly throwing Problems. Therefore, the existing integration tests in BookingTest give us good coverage, as shown below:

![Screenshot 2024-01-30 at 11 44 56](https://github.com/ministryofjustice/hmpps-approved-premises-api/assets/22135634/6ed250f8-336b-4f1c-a973-8bf4dec83945)

Therefore, whilst i agree it would be good to have service level unit tests, I think the coverage we have the moment is sufficient to at least get this change merged as it's blocking some work we need for UI

Reply:
Yes, i agree on establishing strategy for unit vs integration tests ... i'll make a note for the next all-cas API call 👍